src/partition/repart.c

```diff
@@ -7101,6 +7089,34 @@ static int parse_argv(int argc, char *argv[]) {
                         *p = gpt_partition_type_override_architecture(*p, arg_architecture);
         }
 
+        if (private_key) {
+                _cleanup_(erase_and_freep) char *k = NULL;
+                size_t n = 0;
+
+                r = read_full_file_full(
+                                AT_FDCWD, private_key, UINT64_MAX, SIZE_MAX,
+                                READ_FULL_FILE_SECURE|READ_FULL_FILE_WARN_WORLD_READABLE|READ_FULL_FILE_CONNECT_SOCKET,
+                                NULL,
+                                &k, &n);
+                if (r == -ENOENT && strchr(private_key, ':')) {
```

Comment:
urks, i really don't like ambiguities like this whenever we can. We so far managed to do without any ambiguity like this.

I'd prefer if we'd expose this under some non-ambiguous scheme, maybe pair `--private-key=` with a new setting `--private-key-url=` or so? I mean, at least in the pkcs11 case you specify something you'd call an there, don't you? is that generally the case for other relevant modules too? i.e. tpm engine? i have no idea how engines/provides work in practice? just asking?

Reply:
I had a separate parameter for the provider that gated this logic, but @DaanDeMeyer asked me to remove it in the previous PR.

In general, for example in the openssl binary itself, you use the same private-key field that you would use to pass the plaintext key file, so this is consistent with the common approach that I can see.

With openssl you'd do something like: `openssl smime -sign -inkey 'pkcs11:....' -engine pkcs11`

Reply:
I don't mind at all whether there's a new command line parameter or not, so just pick one scheme you both agree with and I'll implement it

Reply:
the openssl cmdline leaves no ambiguity, because you pick the mode explicitly via -engine...

the problem i really have is with the ambiguity. i.e. "pkcs11:foo" could be a filename, could be a url, you can't know. hence i think `--private-key-url=` would be nicest I think, it avoids the ambiguity.

Reply:
I've moved it to --private-key-uri

Reply:
As long as `--private-key-uri` can do everything and you don't have to use either --private-key or --private-key-uri depending on whether you want a file or not I'm OK with it.

Reply:
Lennart wants a separate parameter for files, and a separate parameters for URIs. You want one parameter to accept both. I cannot do both things at the same time.

Reply:
@bluca I mean that we can have both parameters, but `--private-key-uri` should also work if `file:/a/b/c` is specified, which I think will work by default, but it'd be nice if you checked that worked

Reply:
It doesn't work, no idea how to make that work - or do you mean to change this to manually check if the uri starts with file: and if so redirect to the old file-based pre-existing parameter?

Reply:
i mean, we already released this stuff right, requiring file: would be quite a compat break...

I think in future we should maybe follow the rule for similar cases that fs path should be prefixed with "/" or "./", and that strings prefixed with neither are always something else. or in other words: if you explicitly specify a relative path or an absolute path it's a path, otherwise something else.

but here it's kinda too late...

Reply:
oh, i get what @DaanDeMeyer suggests. and I think that would be fine, i.e. manually parse the `file:/` prefix and do the right thing.

That said, parsing `file:/` is kinda nasty. Just look at the ambiguities already listed on the wikipedia page:

https://en.wikipedia.org/wiki/File_URI_scheme

Reply:
https://datatracker.ietf.org/doc/rfc8089/

Reply:
The previous parameter is there and still works, I really don't want to get into parsing that stuff by hand

Reply:
i sympathize. i think the file:// stuff could be added later, if we want to. no need to complicate things with that now.